src/main/java/frc/robot/subsystems/claw/ClawSubsystem.java

```diff
@@ -0,0 +1,131 @@
+package frc.robot.subsystems.claw;
+
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj.Solenoid;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+
+import static frc.robot.Constants.ClawConstants.*;
+
+public class ClawSubsystem extends SubsystemBase {
+  private TalonSRX rightMotor;
+  private TalonSRX leftMotor;
+  private Solenoid pfft;
+
+  // Position variables for the motors
+  private double rightPos;
+  private double leftPos;
+
+  // Right and left motor powers -- state variables
+  private double rightMotorPower;
+  private double leftMotorPower;
+
+  // State variable for whether the pneumatic is lifted or not
+  private boolean liftClaw;
+
+  public ClawSubsystem() {
+    CommandScheduler.getInstance().registerSubsystem(this);
+
+    // Initialize motors and pneumatic
+    rightMotor = new TalonSRX(rightMotorPort);
+    leftMotor = new TalonSRX(leftMotorPort);
+
+    pfft = new Solenoid(pfftPCMPort);
+
+    // Configure the motors for the encoders
+    rightMotor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+    rightMotor.setSelectedSensorPosition(0);
+
+    leftMotor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+    leftMotor.setSelectedSensorPosition(0);
+
+    // Set initial motor position vars (for checking if motors are stalled)
+    rightPos = 0;
+    leftPos = 0;
+
+    liftClaw = false;
+
+  }
+
+  @Override
+  public void periodic() {
+
+    // Set motor powers
+    rightMotor.set(ControlMode.PercentOutput, rightMotorPower);
+    leftMotor.set(ControlMode.PercentOutput, leftMotorPower);
+
+    // Set pneumatic
+    pfft.set(liftClaw);
+  }
+
+  /**
+   * Set motor powers to open.
+   */
+  public void setOpenPowers() {
```

Comment:
how it makes more sense to me is a boolean clawOpen, and then handle stalling logic / powers inside each periodic loop.. for one, it abstracts what should be abstracted in my opinion (I can't imagine anyone, including autonomous, needing something besides open + closed for claw control). that means the commands would just toggle this boolean, and then in periodic we take the boolean, and see which way to power the motors. If the motors are stalling, then we stop powering them.

this also adds the benefit of not running into a scenario where having multiple claw commands 'consumes' the change in position of the encoders, making the stall return true. Essentially what I'm saying is that this logic (for when to move the motors to get to the desired position) should only happen in the subsystem anyways, cuz it's at the subsystem's level of abstraction... idk if that makes sense?

Reply:
Hmm... maybe...

Part of the reason why the code is structured like this is because we discussed last class and decided to have a button "press and hold" control the opening of the claw. So in that sense it seemed more logical to me to create a command that handles that instead of the subsystem. Because then, the subsystem would need more than just one boolean clawOpen to control the claw opening... otherwise there's no way to implement the button hold --> open claw thing.

Reply:
For the second point about consuming the position of encoders.. what exactly does that mean? Like if two commands were trying to figure out if the motors are stalling? Also according to the docs, I believe only one command of a given subsystem can be run at a time. So there should never be any conflicts... idk I might have it totally wrong